Packs/GoogleCloudFunctions/Integrations/GoogleCloudFunctions/GoogleCloudFunctions.py

```diff
@@ -0,0 +1,177 @@
+        if region:
+            self.region = region
+        name = f'projects/{self.project}/locations/{self.region}/functions/{function_name}'
+        body = {'data': data}
+        return self.service.projects().locations().functions().call(name=name, body=body).execute()
+
+
+'''COMMAND FUNCTIONS'''
+
+
+def functions_list_command(client: GoogleClient, args: dict):
+    region = args.get('region')
+    project_id = args.get('project_id')
+    res = client.functions_list(region, project_id)
+    functions = res.get('functions', [])
+    keys = list(functions[0].keys())
+    keys.remove('name')
+    disp_region = 'All' if client.region == '-' else client.region
+    hr = tableToMarkdown(f'Functions in project "{client.project}" and region "{disp_region}"',
+                         functions, ['name'] + keys)
+    ec = {'GoogleCloudFunctions.Function(val.name && val.name == obj.name)': functions}
+    return hr, ec, res
+
+
+def region_list_command(client: GoogleClient, args: dict):
+    project_id = args.get('project_id')
+    res = client.region_list(project_id)
+    regions = res.get('locations', [])
+    hr = tableToMarkdown(f'Regions in project "{client.project}"', regions, ['locationId', 'name', 'labels'])
+    ec = {'GoogleCloudFunctions.Region(val.locationId && val.locationId == obj.locationId)': regions}
+    return hr, ec, res
+
+
+def get_function_by_name_command(client: GoogleClient, args: dict):
+    function_name = args.get('function_name', '')
+    project_id = args.get('project_id')
+    region = args.get('region')
+    res = client.function_by_name(function_name, region, project_id)
+    keys = list(res.keys())
+    keys.remove('name')
+    hr = tableToMarkdown(f'Here are the details for {args.get("function_name")}:', res, ['name'] + keys)
+    ec = {'GoogleCloudFunctions.Function(val.name && val.name == obj.name)': res}
+    return hr, ec, res
+
+
+def execute_function_command(client: GoogleClient, args: dict):
+    project_id = args.get('project_id')
+    region = args.get('region')
+    function_name = args.get('function_name', '')
+    parameters = format_parameters(args.get('parameters', ''))
+    res = client.execute_function(function_name, parameters, region, project_id)
+    hr = tableToMarkdown(f'Execution details for {args.get("function_name")}:', res)
+    ec = {'GoogleCloudFunctions.Execution(val.executionId && val.executionId == obj.executionId)': res}
+    return hr, ec, res
+
+
+'''HELPER FUNCTIONS'''
+
+
+def format_parameters(parameters: str) -> str:
+    final_s = ''
+    for i in parameters.split(','):
+        temp = i.strip()
+        if len(temp.split(':')) != 2:
+            raise ValueError(f"Key:Value pair {i} is not in the correct format.")
+        key, value = temp.split(':')
+        final_s += key.strip() + ':' + value.strip() + ','
+    final_s = final_s[:-1]
+    final_s = final_s.replace(',', '","')
+    final_s = '{"' + final_s.replace(':', '":"') + '"}'
+    return final_s
```

Comment:
what if value includes `"`
could be simplified:
![image](https://user-images.githubusercontent.com/30797606/78502069-ce051d80-7767-11ea-807d-fce12eb6d205.png)